lib/semian/circuit_breaker.rb

```diff
@@ -74,20 +74,20 @@ def destroy
 
     private
 
-    def close
+    def transition_to_close
       log_state_transition(:closed)
       @state.close
       @errors.clear
     end
 
-    def open
+    def transition_to_open
       log_state_transition(:open)
       @state.open
     end
 
-    def half_open
+    def transition_to_half_open
       log_state_transition(:half_open)
-      @state.half_open
+      @state.half_open!
```

Comment:
There's a slight inconsistency here now — `half_open!` gets a bang while `open` and `close` do not.